src/NUnitFramework/nunitlite/TextRunner.cs

```diff
@@ -252,6 +252,17 @@ private int Execute()
 
         #region Helper Methods
 
+        private void LoadTests(IDictionary<string, object> runSettings)
+        {
+            DateTime startTime = DateTime.UtcNow;
+            long startTicks = Stopwatch.GetTimestamp();
+            _runner.Load(_testAssembly, runSettings);
+            DateTime endTime = DateTime.UtcNow;
+            double duration = (double)(Stopwatch.GetTimestamp() - startTicks) / Stopwatch.Frequency;
```

Comment:
I copied our existing logic for start and end of test. At the time I implemented it (which was quite a long time ago) there were known problems with the StopWatch instance and the static method was the only approach that worked reliably. I don't know what a new review of the class would show and I really didn't want to undertake it.

If it seems like a problem (it's the same logic we have used for ages) I can try some tests and correct the elapsed time if it comes out negative.

Reply:
Since it's copy and paste, could we move it to a common helper method? That will also resolve what was bothering me: that this doesn't say what it does, it just does it in a complex way.
I doubt that a test will get it to come out negative unless the system performance counters report in extremely high accuracy and the machine has been running for an unusual amount of time.

It's probably too long ago, but if you recall what the unreliability was, I love to know about that kind of thing.

Reply:
A common method makes sense.

IIRC there were two basic problems: (1) when running on multiple processors, the same processor has to be used for starting and stopping the stopwatch, depending on the BIOS implementation and (2) when there is no hires counter, the system time is used internally, which is a problem because the time can be changed, perhaps even by the test we are running (apparently, this was a "won't fix" bug report to Microsoft in 2012).

Reply:
Hmm... So this is in nunitlite. The other uses are in nunit.framework. We can have a utility method in the framework, which is called by nunitlite. Are we OK with that? Or should we work toward separating nunitlite from relying on helper methods within the framework? I can see doing it either way, since we rebuild nunitlite for each version of the framework. OTOH, if we want to aim toward a more independent nuntlite, we would have to change this at a later time, along with other places where we do something similar.